libraries/chain/controller.cpp

```diff
@@ -304,6 +305,11 @@ struct controller_impl {
       apply_handlers[receiver][make_pair(contract,action)] = v;
    }
 
+   // hotstuff consensus updates these commitments to be included in next produced blocks. returned vector may be empty
+   hs_commitments& get_hs_commitments() {
+      return commitments;
```

Comment:
Where is this set?

Reply:
There is no set. The non-const  returned vector is updated (commitments erased or added) in `chain_pacemaker.hpp`